diff --git a/src/executables/csql.c b/src/executables/csql.c
index 7838b2d46..9a4b4c745 100644
--- a/src/executables/csql.c
+++ b/src/executables/csql.c
@@ -203,7 +203,7 @@ static bool csql_Database_connected = false;
 
 static bool csql_Is_interactive = false;
 static bool csql_Is_sigint_caught = false;
-static bool csql_Is_echo_on = false;
+static bool csql_Is_echo_on = true;
 enum
 { HISTO_OFF, HISTO_ON };
 static int csql_Is_histo_on = HISTO_OFF;
